addons/website_sale/static/src/website_builder/dynamic_snippet_category_option_plugin.js

```diff
@@ -0,0 +1,112 @@
+import { Plugin } from "@html_editor/plugin";
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { DynamicSnippetCategoryOption } from "./dynamic_snippet_category_options";
+import { setDatasetIfUndefined } from "@website/builder/plugins/options/dynamic_snippet_option_plugin";
+import { withSequence } from "@html_editor/utils/resource";
+import { DEVICE_VISIBILITY } from "@website/builder/option_sequence";
+import { MediaDialog } from "@html_editor/main/media/media_dialog/media_dialog";
+import { rpc } from "@web/core/network/rpc";
+
+
+export class DynamicSnippetCategoryOptionPlugin extends Plugin {
+    static id = 'dynamicSnippetCategoryOptionPlugin';
+    selector = "section.s_dynamic_category"
+    resources = {
+        builder_options: [
+            withSequence(DEVICE_VISIBILITY, {
+                OptionComponent: DynamicSnippetCategoryOption,
+                selector: this.selector,
+                groups: ['website.group_website_designer'],
+            }),
+        ],
+        on_snippet_dropped_handlers: this.onSnippetDropped.bind(this),
+    };
+
+    async onSnippetDropped({ snippetEl }) {
+        if (snippetEl.matches(this.selector)) {
+            for (const [optionName, value] of [
+                ['columns', '2'],
+                ['height', 'small'],
+                ['filterId', '0'],
+                ['allProducts', 'true'],
+                ['button', 'Explore Now'],
+                ['alignment', 'left'],
+            ]) {
+                setDatasetIfUndefined(snippetEl, optionName, value);
+            }
+        }
+    }
+}
+
+export class CategoryMediaDialog extends MediaDialog {
+    async save(){
+        rpc('/snippets/category/set_image',{
+            category_id: parseInt(this.props.node.parentElement.dataset),
```

Comment:
I am not able to replace Media for the categories, I hope this will fix the issue.
```suggestion
            category_id: parseInt(this.props.node.parentElement.dataset.categoryId),
```